src/user/profile.js

```diff
@@ -310,6 +311,8 @@ module.exports = function (User) {
 			throw new Error('[[user:change_password_error_privileges]]');
 		}
 
+		await plugins.hooks.fire('filter:password.check', { password: data.newPassword, uid, username, currentPassword: data.currentPassword });
```

Comment:
uid is the uid of the user making the call, data.uid is the user whose password is being changed, so in this case the hook will received the uid and username of the calling user. 
It is fine if the user is changing their own password but if an admin is changing their password the hook gets their data. Might want to pass in data.uid as uid and get the username of data.uid as well

Reply:
Hmm... Thanks for the uid pointer, but about forwarding just uid, the issue is with registration where it's not yet set.
Would it be fine to just forward `userData` in registration and leave password change and reset with just uid? Handling two cases seems easy enough if it comes up, so it's probably fine (I'm not the biggest fan of one hook having two object shapes, but it happens in some other hooks and it'll probably be a rare requirement tbh.).

Reply:
I think it's fine as is now, anyone using these hooks should know what they are doing and handle the uid=0 case and/or userData being there.